galsim/randwalk.py

```diff
@@ -100,21 +117,57 @@ class RandomWalk(GSObject):
     _is_analytic_x = False
     _is_analytic_k = True
 
-    def __init__(self, npoints, half_light_radius, flux=1.0, rng=None, gsparams=None):
+    def __init__(self,
+                 npoints,
+                 half_light_radius=None,
+                 flux=1.0,
+                 profile=None,
+                 rng=None,
+                 gsparams=None):
+
         from .random import BaseDeviate
 
         self._npoints = int(npoints)
-        self._half_light_radius = float(half_light_radius)
-        self._flux = float(flux)
-        self._gsparams = GSParams.check(gsparams)
 
-        # we will verify this in the _verify() method
         if rng is None:
             rng = BaseDeviate()
         self._rng=rng
+
+        self._flux = float(flux)
         self._verify()
 
-        self._set_gaussian_rng()
+        if profile is None:
+            if half_light_radius is None:
+                # use runtime error, because this is a fatal error
+                raise RuntimeError("send a half_light_radius "
+                                   "when not sending a profile")
+            if half_light_radius <= 0.0:
+                raise GalSimRangeError("half light radius must be > 0", half_light_radius, 0.)
+
+            profile=Gaussian(
+                half_light_radius=half_light_radius,
+                flux=flux,
+            )
+
+            self._half_light_radius=half_light_radius
+
+            self._set_gaussian_rng()
+        else:
+            if not isinstance(profile, GSObject):
+                raise TypeError("profile must be a GSObject")
+            # the half light radius is not used
```

Comment:
Do we maybe want to check here that the input `half_light_radius` and `flux` are None?  (I.e. not specified.)

Reply:
done